### PR TITLE
feat: core i18n

### PR DESCRIPTION
**Pull request by jeeyun**
The I18nService is a static class that gives users the ability to use and override strings within the components for internationalization / globalization. One can override default values globally for their application or override per component instance as needed. See https://github.com/vmware/clarity/issues/4502 for more details.

Signed-off-by: Jeeyun Lim <jeeyun.lim@gmail.com>

## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)
- [ ] If applicable, have a visual design approval

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] clarity.design website / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: https://github.com/vmware/clarity/issues/4502

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: packages/core/src/internal/services/i18n.service.ts [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+    nextMonth: 'Next month',
+    previousDecade: 'Previous decade',
+    nextDecade: 'Next decade',
+    currentDecade: 'Current decade',
+    selectMonthText: 'Select month, the current month is {CALENDAR_MONTH}',
+    selectYearText: 'Select year, the current year is {CALENDAR_YEAR}',
+    daypickerSRCurrentMonthPhrase: 'The current month is {CURRENT_MONTH}',
+    daypickerSRCurrentYearPhrase: 'The current year is {CURRENT_YEAR}',
+    daypickerSRCurrentDecadePhrase: 'The current decade is {DECADE_RANGE}',
+  },
```
Comment (jeeyun) [inline packages/core/src/internal/services/i18n.service.ts]: Should I remove all the strings that were carried over from the Angular library and re-add them as needed we implement the components? In particular, the datepicker has these string values where a part of the value is meant to be dynamic. I don't believe we'll need this behavior with this I18nService, so at the very least I expect these values to change. Some of these keys might go away too depending on how we design / implement future components.

Comment (coryrylan) [reply packages/core/src/internal/services/i18n.service.ts]: That sounds good. Re-adding them as needed will give us time to review/audit them as we go so we don't include anything that isn't needed.

File: packages/core/src/alert/alert-group.stories.ts [hunk 2]
```diff
@@ -4,14 +4,18 @@
...
 import { action } from '@storybook/addon-actions';
 import { html } from 'lit-html';
 import { ClarityIcons, headphonesIcon, nodeGroupIcon, timesCircleIcon } from '@clr/core/icon';
 import customElements from '../../dist/core/custom-elements.json';
 
 ClarityIcons.addIcons(headphonesIcon, nodeGroupIcon, timesCircleIcon);
 
+I18nService.localize({
+  alert: { closeButtonAriaLabel: 'close this alert' },
+});
```
Comment (coryrylan) [inline packages/core/src/alert/alert-group.stories.ts:18]: Small nit, we should probably import the `I18nService` from `@clr/core` here just for helping set an example for anyone looking at the example code in storybook

File: packages/core/src/alert/alert-group.stories.ts [hunk 3]
```diff
@@ -43,8 +47,12 @@ export const alertGroup = () => {
   return html`
     <div cds-layout="vertical gap:sm">
       <cds-alert-group status="info" aria-label="This is an example info alert group">
-        <cds-alert closable>
-          This example is an alert a user may be able to close inside an alert group with a status of "info".
+        <cds-alert
+          closable
+          aria-label="This is an example info alert group"
+          cds-i18n='{ "closeButtonAriaLabel": "close my example alert"}'
```
Comment (mathisscott) [inline packages/core/src/alert/alert-group.stories.ts:54]: fwiw, an example that also showed how to change the default string of the alert icon would make accessibility very happy.

**Review by mathisscott — APPROVED** [review]

File: packages/core/src/internal/services/i18n.service.ts [hunk 4]
```diff
@@ -0,0 +1,223 @@
...
+
+  alert: {
+    closeButtonAriaLabel: string;
+    loading: string;
+    info: string;
+    success: string;
+    warning: string;
+    danger: string;
+  };
+  datagrid: {
```
Comment (coryrylan) [inline packages/core/src/internal/services/i18n.service.ts]: Are we still planning on  removing these and only adding them as the component is added?

Comment (jeeyun) [reply packages/core/src/internal/services/i18n.service.ts]: Yup, I will remove all the extra ones that aren't currently used in core.

**Review by coryrylan — APPROVED** [review]

**Pull request merged**